### PR TITLE
Checkbox and RadioButton fixes

### LINKED ISSUE #4003: Checkbox and Radio Button shift when clicked
### Package name(s)

Core (@salt-ds/core)

### Package version(s)

latest

### Description

https://github.com/user-attachments/assets/61e16d9b-bc68-4678-a1ed-c8dd5465e43e

https://github.com/user-attachments/assets/708efb80-9aeb-404f-b5b2-f1a75031938d

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Operating system

- [ ] macOS
- [x] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [x] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [ ] I am an employee of JPMorgan Chase & Co.

### PR DESCRIPTION
**Pull request by joshwooding**
Closes #4003

### CODE REVIEW

File: packages/core/src/checkbox/CheckboxIcon.css [hunk 1]
```diff
@@ -68,6 +68,13 @@
 
 .saltCheckboxIcon-readOnly,
 .saltCheckbox:hover .saltCheckboxIcon-readOnly {
+  background: unset;
```
Comment (origami-z) [inline packages/core/src/checkbox/CheckboxIcon.css:71]: This needs a change mention?

Comment (joshwooding) [reply packages/core/src/checkbox/CheckboxIcon.css:71]: Ah, unintentional. Didn't want to fix this yet

Comment (joshwooding) [inline packages/core/src/checkbox/CheckboxIcon.css:71]:
```suggestion
```

File: packages/core/src/radio-button/RadioButtonIcon.css [hunk 2]
```diff
@@ -54,6 +54,13 @@
 
 .saltRadioButtonIcon-readOnly,
 .saltRadioButton:hover .saltRadioButtonIcon-readOnly {
+  background: unset;
```
Comment (joshwooding) [inline packages/core/src/radio-button/RadioButtonIcon.css:57]:
```suggestion
```

**Review by mark-tate — APPROVED** [review]

**Review by tomhazledine — APPROVED** [review]

**Pull request merged**